### DISCUSSION TITLE
"MET/v12.0.0-beta4 and METPLUS/v6.0.0-beta4" can not find a matched FCST VAR in GEFSv12.3 grib2 output

### DISCUSSION THREAD

**Discussion by Ho-ChunHuang-NOAA** [dtcenter/METplus#2607] [Configuration] [answered]
GEFSv12 grib2 output contains several AOTK records with different aerosol types, aerosol size, and optical wavelength, e.g.,

> 5:4190105:d=2024052712:AOTK:entire atmosphere:57 hour fcst:aerosol=Total Aerosol:aerosol_size <2e-05:aerosol_wavelength >=5.45e-07,<=5.65e-07

The matching condition was created to find AOTK from **record 5** and they were suggested by the DTC, i.e.,
```
FCST_VAR1_NAME = AOTK
FCST_VAR1_LEVELS = A0
FCST_VAR1_OPTIONS = GRIB2_process=62000;GRIB2_ipdtmpl_index = [ 2, 10 ]; GRIB2_ipdtmpl_val = [ 62000, 545 ]
```
and they worked to find matching FCST VAR in met/v11.0.2 and metplus/v5.0.1.  But they failed to find a matching FCST VAR in met/v12.0.0-beta4 and metplus/v6.0.0-beta4.  I do not know when the problem started.  I can only tell from the last stat output that it was using met/v11.0.2.

A simple testing seems to indicate that "FCST_VAR1_OPTIONS = GRIB2_process=62000;GRIB2_ipdtmpl_index = [ 2, 10 ]; GRIB2_ipdtmpl_val = [ 62000, 545 ]" prevents PointStat from finding a matched FCST VAR.

The data used in the PointStat are too large to drop here.  The data, log output, and config files can be found in https://www.emc.ncep.noaa.gov/mmb/hchuang/dtc_discussion.

[PointStat_fcstGEFSAero_obsAERONET_FULL.conf.txt](https://github.com/user-attachments/files/15739899/PointStat_fcstGEFSAero_obsAERONET_FULL.conf.txt)
[evs_gefs_g2o_stats_aeronet_20240527_v11.0.2.txt](https://github.com/user-attachments/files/15739900/evs_gefs_g2o_stats_aeronet_20240527_v11.0.2.txt)
[evs_gefs_g2o_stats_aeronet_20240527_v12.0.0.txt](https://github.com/user-attachments/files/15739901/evs_gefs_g2o_stats_aeronet_20240527_v12.0.0.txt)

**Comment by AliciaBentley-NOAA** [comment]
Thank you for starting this discussion with DTC, @Ho-ChunHuang-NOAA!

**Comment by j-opatz** [comment, marked as answer]
Hi @Ho-ChunHuang-NOAA,

And thank you for your question. The good news is this issue is easily fixed and the explanation is relatively straight forward as well.
The punchline is your usage of `GRIB2_process` is incorrect/unnecessary; your usage of `GRIB2_ipdtmpl_index` and `GRIB2_ipdtmpl_val` is sufficient to pull out the correct line of AOTK. 

To confirm this issue and solution, I went ahead and tested 2 commands of Plot-Data-Plane against both MET 11.0.3 and MET 12.0.0 beta4: one command included the GRIB2_process variable,

```
plot_data_plane gefs.chem.t00z.a2d_0p25.f003.grib2 gefs_test_12.0.0.ps \
'name="AOTK"; level="A0"; GRIB2_process=62000; GRIB2_ipdtmpl_index=[2,10]; GRIB2_ipdtmpl_val=[62000,545];' 
```

and the second command omitted it,

```
plot_data_plane gefs.chem.t00z.a2d_0p25.f003.grib2 gefs_test_12.0.0.ps \
'name="AOTK"; level="A0"; GRIB2_ipdtmpl_index=[2,10]; GRIB2_ipdtmpl_val=[62000,545];'
```

As you discovered, the newest beta of MET (12.0.0-beta4) errors out with the first command, unable to find any matches in the file. This is the *correct* behavior, as `GRIB2_process` is an integer specifying the generating process. [This table](https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table4-3.shtml) shows the recognized integers that can be provided -- note how the maximum value allowed is 255 (far lower than the 62000 you supplied!).

I'm not quite sure how the `GRIB2_process` variable with its setting at 62000 was allowed to function in MET 11.0.3, but I've started an internal discussion with John HG to see if there's anything we need to further complete to make sure the variable continues functioning as it was originally intended.

**Reply by j-opatz** [reply]
@Ho-ChunHuang-NOAA were you able to test this solution out yet? I wanted to make sure we address your issues in a timely manner and close out this Discussions topic when it is sufficiently resolved.

**Comment by Ho-ChunHuang-NOAA** [comment]
@j-opatz  I am able to performed one test and it passed.  I am in a process of setup 2nd test with different input.  Please stay tune for my finding.

**Comment by Ho-ChunHuang-NOAA** [comment]
@j-opatz I have finished my test. The revised "data description";
plot_data_plane gefs.chem.t00z.a2d_0p25.f003.grib2 gefs_test_12.0.0.ps \
'name="AOTK"; level="A0"; GRIB2_ipdtmpl_index=[2,10]; GRIB2_ipdtmpl_val=[62000,545];'
is working in MET (12.0.0-beta4) to find the matching pair of intended FCST VAR.   Thank you for providing the solution.
